### PR TITLE
Added checks for correct format of Date

### PR DESCRIPTION
**Pull request by codepoet2017390**
- Changed Date format to dd-mm-yyyy
- Added checks if date format is wrong

### CODE REVIEW

File: app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    companion object {
+
+        fun checkDateFormat(date: String) : Boolean {
+            return DateUtlis().validate(date)
+        }
+
+        fun getCorrectDateFormat(date:String) : String{
+           val StringArray = date.split("-")
+            var correctFormat: String=""
+            correctFormat= correctFormat+StringArray[2]+"-"+StringArray[1]+"-"+StringArray[0]
```
Comment (isears) [inline app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt]: Do we know for sure that this won't throw an outofbounds excpetion? Maybe add a check to make sure the result of `date.split("-")` is 3 elements long.

File: app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            return correctFormat.trim()
+        }
+
+    }
+
+    fun validate(dateString: String): Boolean {
```
Comment (isears) [inline app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt:22]: Is there a java library that can handle this for us? Seems like a lot of opportunity for error.

**Review by isears — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt (hunk 1 above)
Comment (codepoet2017390) [reply app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt]: Yes actually this function is only called once the date is correctly verified.

File: app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt (hunk 2 above)
Comment (codepoet2017390) [reply app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt:22]: Actually, the in-built classes limit our checks. Many things like if month is greater than 12 or not are not checked by it. Plus I rigorously tested the function.  The best I can do here is put it in a try catch block . That eliminates any possibility of unchecked cases we might have forgotten.

Also this idea of writing the whole code by myself was taken form OpenMRS android client which does something like this only.

Comment (isears) [reply app/src/main/java/com/example/nigeriatelemedicineapp/utils/DateUtlis.kt:22]: Which in-built classes are you using? Can we use the java.text package (https://docs.oracle.com/javase/8/docs/api/java/text/package-summary.html)?

Example here: https://kodejava.org/how-do-i-check-if-a-string-is-a-valid-date/

Let's discuss at our next meeting if you have more questions.

**Review by isears — APPROVED** [review]

**Pull request merged**